msmart/device/AC/command.py

```diff
@@ -116,7 +116,9 @@ def payload(self) -> bytes:
 
         # Build target temp and mode byte
         fractional, integral = math.modf(self.target_temperature)
-        temperature = (int(integral) & 0xF) | (0x10 if fractional > 0 else 0)
+        temperature2 = (int(integral) - 12) & 0x1F
+        temperature = (int(integral) - 16) if integral > 16 else 1

```

Comment:
After reviewing other available references I think it might be proper to bound `temperature` to 17 - 30., and set the fallback value to 0

e.g.
```python
temperature = (int(integral) - 16) if (17 <= int(integral) <= 30) else 0
```

Reply:
Agreed. Without better knowledge I would validate the fallback value against a real device.